pricegraph/src/orderbook.rs

```diff
@@ -548,6 +545,10 @@ mod tests {
         let ReducedOrderbook(orderbook) = orderbook.reduce_overlapping_orders();
         // NOTE: We expect user 1's 2->1 order to be completely filled as well
         // as user 2's 5->3 order and user 4's 6->7 order.
+        // User 3's 7->6 order may be filled or not depending on the order in
+        // which the loop {6, 7} is cleared:
+        // 6->7->6: 100_000 T6 -> 1_000_000 T7 -> 1_000_000 T6, both orders cleared
+        // 7->6->7: 100_000 T7 -> 100_000 T6 -> 1_000_000 T7, only 6->7 cleared
         assert_eq!(orderbook.num_orders(), 6);
```

Comment:
Maybe we should make it `assert!(orderbook.num_orders() == 5 || orderbook.num_orders() == 6);`?